### ISSUE TITLE
GainNode.gain.value no longer reporting accurately?

### ISSUE THREAD

**Issue by TheodoreMichels** [WebAudio/web-audio-api#693]
I'm trying to use the value of a GainNode to control some visuals on a site:
http://www.phylogenetic-composer.com/
I started noticing this problem about a month ago, although the code on the site hasn't changed at all. Basically, each circle in the tree on this page is supposed to expand visually according to its amplitude envelope.

I'm using linearRampToValueAtTime to make a simple AR envelope. Before, when I used gain.value, everything worked as expected. Now, however, I can't make sense of the reported values. As far as I can tell they're very inconsistent. As a result the visuals are very inconsistent as well.

Was the method for querying gain values changed?

Sorry in advance if this issue's already being addressed - I couldn't seem to find it in the open issues.

Thanks for your time.

Here's a simple example to copy into [Canopy](http://hoch.github.io/canopy/) for reference:

        var audioContext = new(window.AudioContext || window.webkitAudioContext)();

        var oscillator = audioContext.createOscillator();
        oscillator.type = "sine";
        oscillator.start();

        var gainNode = audioContext.createGain();
        gainNode.gain.value = 0.0;

        oscillator.connect(gainNode);
        gainNode.connect(audioContext.destination);

        function trigger(){
            gainNode.gain.cancelScheduledValues(audioContext.currentTime);
            gainNode.gain.setValueAtTime(0, audioContext.currentTime);
            gainNode.gain.linearRampToValueAtTime(1.0, audioContext.currentTime + 0.5);
            gainNode.gain.linearRampToValueAtTime(0.0, audioContext.currentTime + 1);
        }
        
        function reportGain(){
            console.log(gainNode.gain.value);
        }
        
        setInterval(trigger, 1000);
        setInterval(reportGain, 50);

**Comment by hoch** [comment]
For one thing, Canopy renders audio with OfflineAudioContext, so getting .value out of AudioParam will behave differently from the real-time audio context. Thus running this example on Canopy is not a great idea.

In theory, the web audio engine will try to keep up with the query as it can, the timing and the accuracy of the value is not guaranteed. You just get what your system can offer. Probably you will have to use AnalyserNode or process the signal with the low pass filter before visualizing it.

#318 is something under the discussion, I believe.

**Comment by TheodoreMichels** [comment]
Thanks for getting back to me so promptly. I should have mentioned that I encountered the same problem in Chrome, Opera, and Firefox, outside of Canopy. However, I'm surprised to report that I just tried the site in Edge and everything appears to be working perfectly...

Regardless, I'll look into the AnalyserNode to see if I can get something that's a bit more consistent across browsers. Thanks for the suggestion, and the reference.
